homeassistant/components/devolo_home_control/__init__.py

```diff
@@ -44,26 +45,29 @@ def shutdown(event: Event) -> None:
         hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, shutdown)
     )
 
-    try:
-        zeroconf_instance = await zeroconf.async_get_instance(hass)
-        entry.runtime_data = []
-        for gateway_id in gateway_ids:
+    zeroconf_instance = await zeroconf.async_get_instance(hass)
+    entry.runtime_data = []
+    offline_gateways = 0
+    for gateway_id in gateway_ids:
+        try:
             entry.runtime_data.append(
                 await hass.async_add_executor_job(
                     partial(
                         HomeControl,
-                        gateway_id=str(gateway_id),
+                        gateway_id=gateway_id,
                         mydevolo_instance=mydevolo,
                         zeroconf_instance=zeroconf_instance,
                     )
                 )
             )
-    except GatewayOfflineError as err:
+        except GatewayOfflineError:
+            offline_gateways += 1
+    if len(gateway_ids) == offline_gateways:
         raise ConfigEntryNotReady(
             translation_domain=DOMAIN,
             translation_key="connection_failed",
-            translation_placeholders={"gateway_id": gateway_id},
-        ) from err
+            translation_placeholders={"gateway_id": ", ".join(gateway_ids)},
+        )
```

Comment:
do we still need the gateway ids, if the are all down? 
maybe update the error message to "Failed to connect to any devolo Home Control central unit."?